src/services/stats/stats-router.ts

```diff
@@ -0,0 +1,127 @@
+                events: { $ne: [] },
+            });
+
+            return res.status(StatusCodes.OK).json({ count: attendees.length });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the number of people eligible for merch item (staff only)
+statsRouter.get(
+    "/merch-item/:PRICE",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const price = req.params.PRICE;
+            if (!price) {
+                return res
+                    .status(StatusCodes.BAD_REQUEST)
+                    .json({ error: "MissingPriceParameter" });
+            }
+            const attendees = await Database.ATTENDEES.find({
+                points: { $gte: price },
+            });
+
+            return res.status(StatusCodes.OK).json({ count: attendees.length });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the number of priority attendees (staff only)
+statsRouter.get(
+    "/priority-attendee",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const currentTime = new Date();
+            const attendees = await Database.ATTENDEES.find({
+                priority_expiry: { $gt: currentTime },
+            });
+
+            return res.status(StatusCodes.OK).json({ count: attendees.length });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the attendance of the past n events (staff only)
+statsRouter.get(
+    "/attendance/:N",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const n = req.params.N;
+            if (!n) {
+                return res
+                    .status(StatusCodes.BAD_REQUEST)
+                    .json({ error: "MissingNParameter" });
+            }
+            const currentTime = new Date();
+            const events = await Database.EVENTS.find({
+                endTime: { $lt: currentTime },
+            })
+                .sort({ endTime: -1 })
+                .limit(parseInt(n));
+
+            const attendanceCounts = events.map(
+                (event) => event.attendanceCount
+            );
+
+            return res
+                .status(StatusCodes.OK)
+                .json({ attendanceCounts: attendanceCounts });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the dietary restriction breakdown/counts (staff only)
+statsRouter.get(
+    "/dietary-restrictions",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const attendees = await Database.ATTENDEES.find({});
+            const dietaryRestrictionsMap: { [key: string]: number } = {};
+
+            attendees.forEach((attendee) => {
+                const dietary_restriction =
+                    attendee.dietary_restrictions as string;
```

Comment:
@AydanPirani question: why did we make dietary restrictions a string and not a string array?

Reply:
Todo: change dietary restrictions to a string array w/ enums
separate dietary restrictions and allergies
for 8 most common allergies, do boolean values (check mark)
add attendee checkin for each day (subobject, arr)

Reply:
and store all items in dietary restrictions / allergies array as all upper case

Reply:
also change endpoints in registration to match it

Reply:
no restrictions, has  ONLY dietary restriction, has ONLY allergies, has BOTH.
^all exclusive